packages/components/src/InputText/InputText.tsx

```diff
@@ -50,14 +61,54 @@ function InputTextInternal(
     },
   }));
 
+  const [currentRows, setRows] = useState(initializeRows());
+
   return (
     <FormField
+      {...props}
       type={props.multiline ? "textarea" : "text"}
       ref={inputRef}
-      {...props}
+      onChange={handleChange}
+      rows={currentRows}
     />
   );
 
+  function initializeRows() {
+    if (!props.rows) return;
+    if (typeof props.rows === "number") {
+      return props.rows;
+    } else {
+      return props.rows.min;
+    }
+  }
+
+  function handleChange(newValue: string) {
+    props.onChange && props.onChange(newValue);
+    if (!props.rows) return;
+    if (
+      typeof props.rows !== "number" &&
+      inputRef &&
+      inputRef.current instanceof HTMLTextAreaElement
+    ) {
```

Comment:
Perhaps?

```suggestion
    if (
      props.rows &&
      typeof props.rows !== "number" &&
      inputRef &&
      inputRef.current instanceof HTMLTextAreaElement
    ) {
```

Reply:
I think this would work as well but linter complains if I do it this way.